src/utils/utils.js

```diff
@@ -100,28 +100,53 @@ function Utils() {
    */
   this.trackHeader = gaUtils.trackEvent('Global Header');
 
+  /**
+   * createFunctionWithTimeout(callback, optTimeout)
+   * The function serves as a pipe to return the function that is passed to it.
+   * It also searves as a timer to execute that function after a certain amount of time.
+   *
+   * @param {Function} callback - The function to be executed after the time of optTimeout
+   * @param {Number} optTimeout
+   * @return {Function}
+   */
+  this.createFunctionWithTimeout = (callback, optTimeout) => {
+  let called = false;
+
+  const fn = () => {
+    if (!called) {
+      called = true;
+      callback();
+    }
+  };
+
+  setTimeout(fn, optTimeout || 500);
+
+  return fn;
+};
+
   /**
    * trackSearchQuerySend = (label ='', dimensions = {})
    * Track a GA click event with custom dimensions.
    * The parameter "dimensions" should be an object with dimensions listed as the following format,
    * { dimensions1: 'value1', dimensions2: 'value2', ... }
+   * This function will send GA event first, and after it is completed, it will trigger the
+   * original event.
    *
    * @param {string} label - Label for GA event.
    * @param {object} dimensions - the object that consists the custom dimensions for the event.
+   * @param {function} hitCallback - the function to be executed after sending GA is completed.
    */
-  this.trackSearchQuerySend = (label = '', dimensions = {}) => {
-    let eventObj = {
-      category: gaConfig.eventCategory,
-      action: gaConfig.eventAction,
+  this.trackSearchQuerySend = (label, dimensions = {}, hitCallback) => {
```

Comment:
No empty string fallback for `label` anymore? Why not?

Reply:
ahh I guess there's no hurt to put it back. The reason is we have done a lot of checks to make sure the value to be passed to `label` is not empty already.